### DISCUSSION TITLE
Enabling ported tools to access z/OS Datasets?

### DISCUSSION THREAD

**Discussion by lbdyck** [zopencommunity/meta#244] [General]
Any plans to add the capability to the ported tools, where appropriate, to access z/OS datasets?

For example cat, less, grep

**Comment by IgorTodorovskiIBM** [comment]
I think there's some overlap with z/OS Open Automation Utilities here.

`dls DATASET | less` - @MikeFultonDev does ZOAU support pager via an environment variable (similar to Git)?
or
You can use ZOAU's `dgrep` instead of grep for dataset. It may be interesting if grep can call out to dgrep if a dataset is passed as an argument (if dgrep is present in the PATH).

**Comment by lbdyck** [comment]
I tried dgrep and couldn't get it to work for me but this is my 1st time using ZOAU.

However - why make ZOAU a pre-req to ZOT?

**Comment by MikeFultonDev** [comment]
@lbdyck just saw this - not sure why dgrep and dcp didn't work for you, but I do think it's reasonable that our 'cp' and 'cat' should 'just work' against datasets like the base UNIX System Services tools do. We have a hack for cp to fail over to /bin/cp and plan to do the same for cat, but it would be good if the community did better (maybe) or alternately if we were able to get z/OS to step up to support the latest options for cp and cat so that we could remove them from our set of ported tools.

**Reply by MikeFultonDev** [reply]
@AnthonyGiorgio separate point if you want to follow up with @lbdyck on dgrep
